### PR TITLE
basic/env-file: also change to state PRE_KEY if we see NEWLINE in sta…

### PR DESCRIPTION
**Pull request by lclgo**
…te COMMENT_ESCAPE

When we see a "\\" in COMMENT state, we change the state to COMMENT_ESCAPE. When we got a new character, we reset the state to COMMENT, but this character is not dispatched. Usually the character is NEWLINE, if so we will stay in COMMENT state until we find the next NEWLINE.

fix: https://github.com/systemd/systemd/issues/27975

### CODE REVIEW

File: src/basic/env-file.c [hunk 1]
```diff
@@ -244,6 +244,10 @@ static int parse_env_file_internal(
 
                 case COMMENT_ESCAPE:
                         state = COMMENT;
+                        if (strchr(NEWLINE, c)) {
+                                state = PRE_KEY;
+                                line++;
+                        }
```
Comment (poettering) [inline src/basic/env-file.c]: now you are setting the state to COMMENT before possible overriding it with PRE_KEY next. maybe move the `state = COMMENT` thing to an else branch to make this clearer, and avoiding the double setting in all cases

**Comment by poettering** [comment]
hmm, in a way this is a compat breakage. not sure if this is reason enough not to do it, in order to bring this in line with shell behaviour.

but please add a NEWS entry for this right-away, making clear this is a (minor) compat breakage.

also, maybe add a log_debug() message about this to the sources when this is triggered, to make this debuggable in case people run into this. just say in the log message that behaviour changed here and how it used to be handled and how it is handled now.

**Comment by poettering** [comment]
would love @keszybz' take on this, whether it's a good idea

**Comment by poettering** [comment]
lgtm for me, but I'd like a final review/merge from @keszybz who knows shells and stuff much better than I do.

**Comment by keszybz** [comment]
Yeah, I think we should merge this. Our docs don't say anything either way (in systemd.syntax(7)), and the old behaviour was an accident of implementation and didn't make much sense.

**Review by keszybz — APPROVED** [review]

**Pull request merged**